packages/date-picker/test/keyboard-input.common.js

```diff
@@ -339,93 +339,6 @@ describe('keyboard', () => {
         await sendKeys({ press: 'Escape' });
         expect(document.activeElement).to.equal(input);
       });
-
-      it('should revert input value on input Esc when empty', async () => {
-        await sendKeys({ type: '1/2/2000' });
-        await nextRender(datePicker);
-
-        await sendKeys({ press: 'Escape' });
-        expect(input.value).to.equal('');
-      });
-
-      it('should not change value on input Esc when empty', async () => {
-        await sendKeys({ type: '1/2/2000' });
-        await nextRender(datePicker);
-
-        await sendKeys({ press: 'Escape' });
-        expect(datePicker.value).to.equal('');
-      });
-
-      it('should revert input value on calendar Esc when empty', async () => {
-        await sendKeys({ type: '1/2/2000' });
-
-        // Move focus to the calendar
-        await sendKeys({ press: 'ArrowDown' });
-        await nextRender(datePicker);
-
-        await sendKeys({ press: 'Escape' });
-
-        expect(input.value).to.equal('');
-      });
-
-      it('should not change value on calendar Esc when empty', async () => {
-        await sendKeys({ type: '1/2/2000' });
-
-        // Move focus to the calendar
-        await sendKeys({ press: 'ArrowDown' });
-        await nextRender(datePicker);
-
-        await sendKeys({ press: 'Escape' });
-
-        expect(datePicker.value).to.equal('');
-      });
-    });
-
-    describe('with value', () => {
-      beforeEach(async () => {
-        datePicker.value = '2000-01-01';
-        await open(datePicker);
-        input.select();
-      });
-
-      it('should revert input value on input Esc when value is set', async () => {
-        // Replace input with a new one
-        await sendKeys({ type: '1/2/2000' });
-        await sendKeys({ press: 'Escape' });
-        expect(input.value).to.equal('1/1/2000');
-      });
-
-      it('should not change value on input Esc when value is set', async () => {
-        // Replace input with a new one
-        await sendKeys({ type: '1/2/2000' });
-
-        await sendKeys({ press: 'Escape' });
-        expect(datePicker.value).to.equal('2000-01-01');
-      });
-
-      it('should revert input value on calendar Esc when value is set', async () => {
```

Comment:
**note:** It's not necessary to test the revert logic on calendar Escape in addition to input Escape. Whenever the overlay is open, the overlay Escape logic takes priority over the input one.